crates/bevy_pbr/src/meshlet/from_mesh.rs

```diff
@@ -0,0 +1,98 @@
+use super::asset::{Meshlet, MeshletBoundingSphere, MeshletMesh};
+use bevy_render::{
+    mesh::{Indices, Mesh},
+    render_resource::PrimitiveTopology,
+};
+use meshopt::{build_meshlets, compute_meshlet_bounds_decoder, VertexDataAdapter};
+use std::borrow::Cow;
+
+impl MeshletMesh {
+    /// Process a [`Mesh`] to generate a [`MeshletMesh`].
+    ///
+    /// This process is very slow, and should be done ahead of time, and not at runtime.
+    ///
+    /// This function requires the `meshlet_processor` cargo feature.
+    ///
+    /// The input mesh must:
+    /// 1. Use [`PrimitiveTopology::TriangleList`]
+    /// 2. Use indices
+    /// 3. Have the exact following set of vertex attributes: `{POSITION, NORMAL, UV_0, TANGENT}`
+    pub fn from_mesh(mesh: &Mesh) -> Result<Self, MeshToMeshletMeshConversionError> {
+        // Validate mesh format
+        if mesh.primitive_topology() != PrimitiveTopology::TriangleList {
```

Comment:
Maybe a method on `Mesh` that could convert to a triangle list would be useful. Not for lines and such, but triangle fan, triangle strip, and whatnot.

Reply:
Sure, not this PR :)